library/std/src/sys/thread_local/os.rs

```diff
@@ -95,8 +148,15 @@ impl<T: 'static> Storage<T> {
             return ptr::null();
         }
 
-        let value = Box::new(Value { value: i.and_then(Option::take).unwrap_or_else(f), key });
-        let ptr = Box::into_raw(value);
+        // Manually allocate with the requested alignment
+        let layout = Layout::new::<Value<T>>().align_to(ALIGN).unwrap();
+        let ptr: *mut Value<T> = (unsafe { crate::alloc::alloc(layout) }).cast();
+        if ptr.is_null() {
+            crate::alloc::handle_alloc_error(layout);
+        }
+        unsafe {
+            ptr.write(Value { value: i.and_then(Option::take).unwrap_or_else(f), key });
+        }
```

Comment:
I think `ptr` is going to be leaked here if get/set panic -- can that happen? It looks like those have a quite a bit of code that they're running...

Reply:
Good catch, should be better now